department-of-veterans-affairs/caseflow
# MattT/APPEALS-7703: Resolve CAMO Case Timeline Issues

Resolves [APPEALS-7703](https://vajira.max.gov/browse/APPEAL-7703)

### Description
I've altered [the logic used in assembling instructions arrays](https://github.com/department-of-veterans-affairs/caseflow/blob/7a9544c8f83149272e06d667d2773b9ab472846a/client/app/queue/components/CompleteTaskModal.jsx#L394) to resolve a few inconsistencies in how task instructions were being displayed on the Case Timeline on the Case Details page for appeals.

### Acceptance Criteria
- [ ] "CAMO Notes" portion only appears once when appeal is routed directly from CAMO to BVA Intake (having circumvented the Program Office phase of the workflow).
- [ ] "Status" and "CAMO Notes" sections which are both entered by CAMO appear together, in addition to "Program Office Notes" and "Detail" which are both entered by the Program Office, should also appear together.

### Testing Plan
1. Intake an appeal with a VHA non-caregiver issue.

2. I would recommend saving the appeal into a variable within the Rails console:
```ruby
a = Appeal.find_by_uuid("<UUID>")
```

3. Switch to a CAMO user (ex: `CAMOUSER`). I would suggest that you add this user to a VHA Program Office for ease of testing:

```ruby
Organization.find_by(name: "Prosthetics").add_user(User.find_by_css_id("CAMOUSER"))
```

4. Navigate to the Case Details page for the appeal you've processed through Intake.

`/queue/appeals/<UUID>`

5. Select the "Assign to Program Office" task action from the actions dropdown. Select whichever program office you've added your user to. Add some text to the `textarea` field (ex: `CAMO to PO`).
Submit the form.

6. Navigate back to the Case Details page. Select "Ready for Review" from the task actions dropdown. Select any radio field. Enter some text into the `textarea` field (ex: `PO to CAMO`). Submit the form.

7. Go back to the Case Details page. Select "Send to Board Intake" from the actions dropdown this time. Select any radio field and once again add some text to the `textarea` field (ex: `CAMO to BVA Intake`). Submit this form too.

8. Go back to the Case Details page and, and this time scroll to the bottom where the Case Timeline section is located. Click on the "View task instructions" link for both the `VhaDocumentSearchTask` and `AssessDocumentationTask`. Ensure that the contents of the instructions sections meet expectations (see screenshots below for reference).

9. Your task tree will now look something like this:

```ruby
a.treee
=>
                                        ┌──────────────────────────────────────────────────────────────────────────┐
Appeal 1632 (D 220731-1632 Original) ── │ ID   │ STATUS    │ ASGN_BY  │ ASGN_TO          │ UPDATED_AT              │
└── RootTask                            │ 7130 │ on_hold   │          │ Bva              │ 2022-08-09 12:35:11 UTC │
    └── PreDocketTask                   │ 7131 │ assigned  │          | BvaIntake        │ 2022-08-16 13:31:00 UTC │
        └── VhaDocumentSearchTask       │ 7152 │ completed │ BVADWISE │ VhaCamo          │ 2022-08-16 13:31:40 UTC │
            └── AssessDocumentationTask │ 7153 │ completed │ CAMOUSER │ VhaProgramOffice │ 2022-08-16 13:31:40 UTC │
                                        └──────────────────────────────────────────────────────────────────────────┘
```

Delete the `VhaDocumentSearchTask` and `AssessDocumentationTask` off of the appeal:
**Note**: These task IDs will almost certainly differ for you. Please refer to your own task tree.

```ruby
# Delete AssessDocumentationTask first
Task.find(7153).delete
Task.find(7152).delete
```

Create a fresh `VhaDocumentSearchTask`:
```ruby
VhaDocumentSearchTask.create!(appeal:a, parent: Task.find(<PreDocketTask-ID>), assigned_to: VhaCamo.singleton, assigned_by: BvaIntake.singleton.users.first)
```

10. Refresh the Case Details page. Select "Send to Board Intake" from the actions dropdown. Select any radio option, and add some notes to the `textarea` field (ex: `CAMO directly to BVA Intake`). Submit the form.

11. Go back to the Case Details page, and scroll down to the Case Timeline. Collapse the task instructions for the `VhaDocumentSearchTask` by clicking on the "View task instructions" link. Ensure that `CAMO Notes:` only shows up once. Refer to the screenshots below for further reference.


### User Facing Changes

 SCENARIO|BEFORE|AFTER
 ---|---|---
 CAMO to Intake|<img width="311" alt="camo-to-intake-pre-changes" src="https://user-images.githubusercontent.com/99351305/184889242-0bd1aa08-d305-4d22-9af3-aab7c9b12ea8.PNG">|<img width="373" alt="camo-to-intake-post-changes" src="https://user-images.githubusercontent.com/99351305/184891693-1efb7a75-3c94-4afa-a112-e27955a7849a.PNG">
 CAMO to PO to CAMO to Intake|![camo-to-po-to-camo-to-intake-pre-changes](https://user-images.githubusercontent.com/99351305/184889289-23f5dd46-213e-4122-b7e0-e610bfae043d.png)|![camo-to-po-to-camo-to-intake-post-changes](https://user-images.githubusercontent.com/99351305/184889149-ea726887-64c0-4761-83e2-05db8f9b2695.png)

### Code Documentation Updates
